ibm/resource_ibm_is_subnet_reserved_ip.go

```diff
@@ -0,0 +1,287 @@
+			},
+			isReservedIPCreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time that the reserved IP was created.",
+			},
+			isReservedIPhref: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL for this reserved IP.",
+			},
+			isReservedIPOwner: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The owner of a reserved IP, defining whether it is managed by the user or the provider.",
+			},
+			isReservedIPType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource type.",
+			},
+		},
+	}
+}
+
+// resourceIBMISReservedIPCreate Creates a reserved IP given a subnet ID
+func resourceIBMISReservedIPCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	// Getting the subnet ID
+	subnetID := d.Get(isSubNetID).(string)
+
+	options := sess.NewCreateSubnetReservedIPOptions(subnetID)
+
+	// Getting the name of the reserved IP if given
+	nameStr := ""
+	if name, ok := d.GetOk(isReservedIPName); ok {
+		nameStr = name.(string)
+	}
+	if nameStr != "" {
+		options.Name = &nameStr
+	}
+
+	// Setting the auto delete
+	autoDeleteBool := false
+	if autoDelete, ok := d.GetOk(isReservedIPAutoDelete); ok {
+		autoDeleteBool = autoDelete.(bool)
+	}
+	options.AutoDelete = &autoDeleteBool
+
+	// Now create the reserved IP
+	rip, response, err := sess.CreateSubnetReservedIP(options)
+	if err != nil || response == nil {
+		return fmt.Errorf("Error creating the reserved IP: %s\n%s", err, response)
+	} else {
+		msg := fmt.Sprintf("Created reserved IP for Subnet id: %s", subnetID)
+		fmt.Println(" ✅ " + "\033[35m" + msg + "\033[0m")
+	}
+
+	// Set id for the reserved IP
+	reservedIPID := *rip.ID
+	d.SetId(fmt.Sprintf("%s/%s", subnetID, reservedIPID))
+
+	// Finally call read method to read the resouce, set the variables and return
+	return resourceIBMISReservedIPRead(d, meta)
+}
+
+func resourceIBMISReservedIPRead(d *schema.ResourceData, meta interface{}) error {
+	rip, err := get(d, meta)
+	if err != nil {
+		return err
+	}
+
+	allIDs, err := idParts(d.Id())
+	subnetID := allIDs[0]
+
+	if rip != nil {
+		d.Set(isReservedIPAddress, *rip.Address)
+		d.Set(isReservedIP, *rip.ID)
+		d.Set(isSubNetID, subnetID)
+		d.Set(isReservedIPAutoDelete, *rip.AutoDelete)
+		d.Set(isReservedIPCreatedAt, (*rip.CreatedAt).String())
+		d.Set(isReservedIPhref, *rip.Href)
+		d.Set(isReservedIPName, *rip.Name)
+		d.Set(isReservedIPOwner, *rip.Owner)
+		d.Set(isReservedIPType, *rip.ResourceType)
+	}
+	return nil
+}
+
+func resourceIBMISReservedIPUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	name := "" // Primarily name is something we are patching
+
+	if d.HasChange(isReservedIPName) {
+		name = d.Get(isReservedIPName).(string)
```

Comment:
since you can update only name you can move all these logic to if d.HasChange(isReservedIPName)